spyder/plugins/projects/plugin.py

```diff
@@ -263,6 +283,91 @@ def on_main_menu_available(self):
             menu_id=ApplicationMenus.Projects,
             section=ProjectsMenuSections.Extras)
 
+    @on_plugin_teardown(plugin=Plugins.Editor)
+    def on_editor_teardown(self):
+        self.editor = self.get_plugin(Plugins.Editor)
+        widget = self.get_widget()
+        treewidget = widget.treewidget
+
+        treewidget.sig_open_file_requested.disconnect(self.editor.load)
+        treewidget.sig_removed.disconnect(self.editor.removed)
+        treewidget.sig_tree_removed.disconnect(self.editor.removed_tree)
+        treewidget.sig_renamed.disconnect(self.editor.renamed)
+        treewidget.sig_tree_renamed.disconnect(self.editor.renamed_tree)
+        treewidget.sig_module_created.disconnect(self.editor.new)
+        treewidget.sig_file_created.disconnect(self._editor_new)
+
+        self.sig_project_loaded.disconnect(self._editor_setup_files)
+        self.sig_project_closed[bool].disconnect(self._editor_setup_files)
+        self.editor.set_projects(None)
+        self.sig_project_loaded.disconnect(self._editor_path)
+        self.sig_project_closed.disconnect(self._editor_unset_path)
+
+        self._editor_new = None
+        self._editor_setup_files = None
+        self._editor_path = None
+        self._editor_unset_path = None
+        self.editor = None
+
+    @on_plugin_teardown(plugin=Plugins.Completions)
+    def on_completions_teardown(self):
+        self.completions = self.get_plugin(Plugins.Completions)
+
+        self.completions.sig_stop_completions.disconnect(
+            self.stop_workspace_services)
+
+        self.sig_project_loaded.disconnect(self._addition_path_update)
+        self.sig_project_closed.disconnect(self._deletion_path_update)
+
+        self._addition_path_update = None
+        self._deletion_path_update = None
+        self.completions = None
+
+    @on_plugin_teardown(plugin=Plugins.IPythonConsole)
+    def on_ipython_console_teardown(self):
+        self.ipyconsole = self.get_plugin(Plugins.IPythonConsole)
+        widget = self.get_widget()
+        treewidget = widget.treewidget
+
+        treewidget.sig_open_interpreter_requested.disconnect(
+            self.ipyconsole.create_client_from_path)
+        treewidget.sig_run_requested.disconnect(self._ipython_run_script)
+
+        self._ipython_run_script = None
+        self.ipyconsole = None
+
+    @on_plugin_teardown(plugin=Plugins.OutlineExplorer)
+    def on_outline_explorer_teardown(self):
+        outline_explorer = self.get_plugin(Plugins.OutlineExplorer)
+        self.sig_project_loaded.disconnect(self._outline_update)
+        self.sig_project_closed.disconnect(self._outline_update)
+        self._outline_update = None
+
+    @on_plugin_teardown(plugin=Plugins.MainMenu)
+    def on_main_menu_teardown(self):

```

Comment:
I think we should remove the entire `Projects` menu, instead doing it by pieces, which (I'd say) still leaves the menu in the interface.

I don't know if that's possible though.